flask_app/models.py

```diff
@@ -1,6 +1,41 @@
 from .app import app
 from flask.ext.sqlalchemy import SQLAlchemy
+from flask.ext.security import Security, SQLAlchemyUserDatastore, \
+    UserMixin, RoleMixin
 
 db = SQLAlchemy(app)
 
 ### Add models here
+
+roles_users = db.Table('roles_users',
+    db.Column('user_id', db.Integer(), db.ForeignKey('user.id')),
+    db.Column('role_id', db.Integer(), db.ForeignKey('role.id')))
+
+
+class Role(db.Model, RoleMixin):
+    id = db.Column(db.Integer(), primary_key=True)
+    name = db.Column(db.String(80), unique=True)
+    description = db.Column(db.String(255))
+
+
+class User(db.Model, UserMixin):
+    id = db.Column(db.Integer, primary_key=True)
+    email = db.Column(db.String(255), unique=True)
+    password = db.Column(db.String(255))
+    active = db.Column(db.Boolean())
+    confirmed_at = db.Column(db.DateTime())
+    roles = db.relationship('Role', secondary=roles_users,
+                            backref=db.backref('users', lazy='dynamic'))
+
+
+# Setup Flask-Security
+user_datastore = SQLAlchemyUserDatastore(db, User, Role)
+security = Security(app, user_datastore)
+
+# Create a user to test with
+@app.before_first_request
+def create_user():
+    db.create_all()
```

Comment:
testserver can be solved in a relatively elegant way - when the TESTING config is on in Flask, just allow any login or a special login (test/test?). In production this is more complex. I think a nice compromise would be a /setup route to ask the user for the admin credentials to be created if there are no users currently. We can keep it very simple (not even bootstrap or anything - that way the user will be able to customize it later).

Also as long as routes are not decorated with requires_login, there should be no need for users if the user doesn't want them.